### PR TITLE
{Core} Stop using vendored SDK in core

### PR DESCRIPTION
**Pull request by jiasli**
**Description**

Stop using vendored subscription SDK in core as [`azure-mgmt-resource`](https://pypi.org/project/azure-mgmt-resource/) has been migrated to Track 2 (https://github.com/Azure/azure-cli/pull/17783).

### CODE REVIEW

**Comment by yonzhan** [comment]
Core

File: src/azure-cli-core/azure/cli/core/_profile.py [hunk 1]
```diff
@@ -75,7 +75,7 @@
 
 _AZ_LOGIN_MESSAGE = "Please run 'az login' to setup account."
 
-_USE_VENDORED_SUBSCRIPTION_SDK = True
+_USE_VENDORED_SUBSCRIPTION_SDK = False
```
Comment (zhoxing-ms) [inline src/azure-cli-core/azure/cli/core/_profile.py:78]: Since the vendor SDK has been deleted, why do we still keep the logic of this switch?

Comment (jiasli) [reply src/azure-cli-core/azure/cli/core/_profile.py:78]: In case there is a track 3 someday. 😏

Comment (zhoxing-ms) [reply src/azure-cli-core/azure/cli/core/_profile.py:78]: 😂😂

**Review by zhoxing-ms — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**